client/consensus/manual-seal/src/lib.rs

```diff
@@ -94,7 +94,7 @@ pub fn import_queue<Block, Transaction>(
 }
 
 /// Params required to start the instant sealing authorship task.
-pub struct ManualSealParams<B: BlockT, BI, E, C: ProvideRuntimeApi<B>, A: ChainApi, SC, CS, CIDP> {
+pub struct ManualSealParams<B: BlockT, BI, E, C: ProvideRuntimeApi<B>, TP, SC, CS, CIDP> {
```

Comment:
Should TP have a trait bound here?

Reply:
If it doesn't need a trait bound in the struct definition I think it's fine IMO. Sometimes it makes error messages clearer but in this case I wouldn't worry.

Reply:
yes it should, it's used here:
https://github.com/paritytech/substrate/blob/62939f986f44c638ac61a34438b8147685298fa5/client/consensus/manual-seal/src/lib.rs#L136